docs/development.md

```diff
@@ -78,4 +78,76 @@ define the WATCH_NAMESPACE environment variable.
 
 ```
 WATCH_NAMESPACE=3scale make run
-```
\ No newline at end of file
+```
+
+## Publishing a new release
```

Comment:
Maybe this doc should be placed on a new file `release.md`? I think it deserves its own doc, separated for real development which implies deploying the operator locally without even generating a new image?

Reply:
Fixed in https://github.com/3scale/saas-operator/commit/5aad4ef9a9d3e1d770aea0ca18d76ca9d392ba87